In [1]:
from pathlib import Path
from keyword_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"Related keywords score: {scores["related_keyword_score"] * 100:.2f}%")

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3794.20it/s]


models/vanilla-scibert
Full match score: 43.68%
Related keywords score: 5.86%



Loading weights: 100%|██████████| 310/310 [00:00<00:00, 3957.64it/s]
Loading weights: 0it [00:00, ?it/s]
Qwen3Model LOAD REPORT from: models/vanilla-qwen
Key                                                                     | Status     | 
------------------------------------------------------------------------+------------+-
base_model.model.layers.{0...27}.mlp.down_proj.lora_A.default.weight    | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.o_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.q_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.gate_proj.lora_B.default.weight    | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.gate_proj.lora_A.default.weight    | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.v_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.v_proj.lora_A.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.o_proj.lora

models/vanilla-qwen
Full match score: 48.22%
Related keywords score: 5.96%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3727.29it/s]


models/jargon-layman-scibert-0.75
Full match score: 45.55%
Related keywords score: 6.06%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3918.62it/s]


models/jargon-layman-scibert-1.00
Full match score: 45.95%
Related keywords score: 6.12%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3554.10it/s]


models/jargon-layman-scibert-0.25
Full match score: 45.65%
Related keywords score: 5.99%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3727.79it/s]


models/jargon-layman-scibert-0.50
Full match score: 46.05%
Related keywords score: 6.05%



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 18217.00it/s]
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.


allenai/scibert_scivocab_uncased
Full match score: 25.00%
Related keywords score: 3.50%


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 11.46% | 1.94% |
| vanilla-scibert | 43.68% | 5.86% |
| jargon-layman-scibert | <u>45.95%</u> | **6.12%** |
| vanilla-qwen | **48.22%** | <u>5.96%</u> |

In [ ]:
import pandas as pd

In [ ]:
scores_for_select_model = get_scores("models/jargon-layman-finetuned-1.00")
search_results: list[list[dict[str, float]]] = scores_for_select_model["search_results"]
search_df: pd.DataFrame = scores_for_select_model["search_df"]

In [ ]:
search_results

In [ ]:
search_df

In [ ]:
results_df = pd.DataFrame(data={
    "query_index": range(len(search_results)),
    "result_id": [
        [result["corpus_id"] for result in single_results]
        for single_results in search_results
    ]
}).explode("result_id")
results_df.head()

In [ ]:
results_df["layman"] = results_df["query_index"].apply(lambda i: search_df["layman"].iloc[i])
results_df["jargon"] = results_df["result_id"].apply(lambda i: search_df["jargon"].iloc[i])
results_df.head()

In [ ]:
search_results[0][0]